Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -363,19 +757,157 @@ void SkylanderPortalWindow::UpdateEdits()
+    if (character.game == IOS::HLE::USB::SWAP_FORCE)
+      pass = true;
+  }
+  if (m_game_filters[3]->isChecked())
+  {
+    if (character.game == IOS::HLE::USB::TRAP_TEAM)
+      pass = true;
+  }
+  if (m_game_filters[4]->isChecked())
+  {
+    if (character.game == IOS::HLE::USB::SUPERCHARGERS)
+      pass = true;
   }
+  if (pass == false)
+    return false;
 
-  return false;
+  if (!name.contains(m_sky_search->text(), Qt::CaseInsensitive))
+    return false;
+
+  switch (GetElementRadio())
+  {
+  case 1:
+    if (character.element != IOS::HLE::USB::MAGIC)
+      return false;
+    break;
+  case 2:
+    if (character.element != IOS::HLE::USB::WATER)
+      return false;
+    break;
+  case 3:
+    if (character.element != IOS::HLE::USB::TECH)
+      return false;
+    break;
+  case 4:
+    if (character.element != IOS::HLE::USB::FIRE)
+      return false;
+    break;
+  case 5:
+    if (character.element != IOS::HLE::USB::EARTH)
+      return false;
+    break;
+  case 6:
+    if (character.element != IOS::HLE::USB::LIFE)
+      return false;
+    break;
+  case 7:
+    if (character.element != IOS::HLE::USB::AIR)
+      return false;
+    break;
+  case 8:
+    if (character.element != IOS::HLE::USB::UNDEAD)
+      return false;
+    break;
+  case 9:
+    if (character.element != IOS::HLE::USB::E_OTHER)
+      return false;
+    break;
+  }
+
+  return true;
 }
 
-void SkylanderPortalWindow::closeEvent(QCloseEvent* event)
+QString SkylanderPortalWindow::GetFilePath(u16 id, u16 var)
 {
-  hide();
+  QDir collection = QDir(m_collection_path);
+  auto& system = Core::System::GetInstance();
+  QString file_path;
+  for (auto file : collection.entryInfoList(QStringList(tr("*.sky"))))
+  {
+    File::IOFile sky_file(file.filePath().toStdString(), "r+b");
+    if (!sky_file)
+    {
+      continue;
+    }
+    std::array<u8, 0x40 * 0x10> file_data;
+    if (!sky_file.ReadBytes(file_data.data(), file_data.size()))
+    {
+      continue;
+    }
+    auto ids = system.GetSkylanderPortal().CalculateIDs(file_data);
+    if (ids.first == id && ids.second == var)
+    {
+      return file.filePath();
+    }
+  }
+  return file_path;
+}
+
+u8 SkylanderPortalWindow::GetCurrentSlot()
+{
+  for (auto radio : m_slot_radios)
+  {
+    if (radio->isChecked())
+    {
+      return radio->property("id").toInt();
+    }
+  }
+  return 0;
+}
+
+int SkylanderPortalWindow::GetElementRadio()
+{
+  for (auto radio : m_element_filter)
+  {
+    if (radio->isChecked())
+    {
+      return radio->property("id").toInt();
+    }
+  }
+  return -1;
+}
+
+QBrush SkylanderPortalWindow::GetBaseColor(std::pair<const u16, const u16> ids)
+{
+  switch (IOS::HLE::USB::list_skylanders.at(ids).game)
+  {
+  case IOS::HLE::USB::SPYROS_ADV:
+    return QBrush(QColor(240, 255, 240, 255));
+    break;
+  case IOS::HLE::USB::GIANTS:
+    return QBrush(QColor(255, 240, 215, 255));
+    break;
+  case IOS::HLE::USB::SWAP_FORCE:
+    return QBrush(QColor(240, 245, 255, 255));
+    ;
```

Comment:
unnecessary

Reply:
Are you referring to the break statements or the background color? I just removed the breaks but the color change seems useful to me- from a new player's perspective, it's impossible to tell which game each character is from from its name alone. For example, if you have the checkboxes checked up through Trap Team, there are 6 versions of Stealth Elf that a new player couldn't tell apart. The colors provide a way to tell which version each one was released.

Reply:
the semicolon on a new line